routes/api/project.rb

```diff
@@ -5,7 +5,7 @@ class CloverApi
     @serializer = Serializers::Api::Project
 
     r.get true do
-      projects = Project.authorized(@current_user.id, "Project:view").all
+      projects = Project.authorized(@current_user.id, "Project:view").all.filter(&:visible)
```

Comment:
We need to add same checks to `routes/web/project.rb`
```suggestion
      projects = Project.authorized(@current_user.id, "Project:view").where(visible: true).all
```